python/wikibot/dashboard.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Class defining a dashboard of the status of Sling updates to WikiData."""
+
+import pywikibot
+import sling
+import sling.flags as flags
+import glob
+
+flags.define("--test",
+             help="use test record file",
+             default=False,
+             action='store_true')
+
+precision_map = {
+  sling.MILLENNIUM: pywikibot.WbTime.PRECISION['millenia'],
+  sling.CENTURY: pywikibot.WbTime.PRECISION['century'],
+  sling.DECADE: pywikibot.WbTime.PRECISION['decade'],
+  sling.YEAR: pywikibot.WbTime.PRECISION['year'],
+  sling.MONTH: pywikibot.WbTime.PRECISION['month'],
+  sling.DAY: pywikibot.WbTime.PRECISION['day']
+}
+
+class Dashboard:
```

Comment:
We usually use "dashboard" about something with a user interface. What about renaming this to something like WikiMonitor?

Reply:
Agree. Done